app/models/service_parameters_set.rb

```diff
@@ -3,5 +3,5 @@ class ServiceParametersSet < ApplicationRecord
   belongs_to :source
   belongs_to :service_offering
 
-  has_many   :service_instances, :dependent => :nullify
```

Comment:
@agrare I've changed this to on delete cascade, I think a plan can't live without offering

Reply:
Possibly?  They are a separate top-level collection though so deletion should probably be handled by refresh.